gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/test/java/io/gravitee/rest/api/service/impl/upgrade/upgrader/IntegrationPrimaryOwnerUpgraderTest.java

```diff
@@ -0,0 +1,270 @@
+            integrations.add(new Integration().toBuilder().id(INTEGRATION_ID).build());
+        }
+        Pageable pageablePage0 = new PageableBuilder().pageNumber(0).pageSize(PAGE_SIZE).build();
+        Pageable pageablePage1 = new PageableBuilder().pageNumber(1).pageSize(PAGE_SIZE).build();
+        when(environmentRepository.findAll()).thenReturn(Set.of(environment()));
+        when(integrationRepository.findAllByEnvironment(ORG_ID, pageablePage0))
+            .thenReturn(new Page<>(integrations.subList(0, 100), 0, 100, 102));
+        when(integrationRepository.findAllByEnvironment(ORG_ID, pageablePage1))
+            .thenReturn(new Page<>(integrations.subList(100, 102), 0, 2, 102));
+        when(apiRepository.search(any(), any())).thenReturn(List.of(api()));
+        when(
+            roleRepository.findByScopeAndNameAndReferenceIdAndReferenceType(
+                RoleScope.API,
+                SystemRole.PRIMARY_OWNER.name(),
+                ORG_ID,
+                RoleReferenceType.ORGANIZATION
+            )
+        )
+            .thenReturn(Optional.ofNullable(role(API_PRIMARY_OWNER_ROLE_ID)));
+        when(
+            roleRepository.findByScopeAndNameAndReferenceIdAndReferenceType(
+                RoleScope.INTEGRATION,
+                SystemRole.PRIMARY_OWNER.name(),
+                ORG_ID,
+                RoleReferenceType.ORGANIZATION
+            )
+        )
+            .thenReturn(Optional.ofNullable(role(INTEGRATION_PRIMARY_OWNER_ROLE_ID)));
+        when(membershipRepository.findByReferencesAndRoleId(MembershipReferenceType.API, List.of(API_ID), API_PRIMARY_OWNER_ROLE_ID))
+            .thenReturn(Set.of(membership()));
+
+        boolean success = upgrader.upgrade();
+        assertThat(success).isTrue();
+
+        verify(membershipRepository, times(102)).create(any());
+    }
+
+    @Test
+    public void upgrade_should_fail_if_no_primary_owner_role_exist_for_integration() throws TechnicalException {
+        when(environmentRepository.findAll()).thenReturn(Set.of(environment()));
+        when(integrationRepository.findAllByEnvironment(any(), any())).thenReturn(new Page<>(List.of(integration()), 0, 1, 1));
+        when(apiRepository.search(any(), any())).thenReturn(List.of(api()));
+        when(
+            roleRepository.findByScopeAndNameAndReferenceIdAndReferenceType(
+                RoleScope.API,
+                SystemRole.PRIMARY_OWNER.name(),
+                ORG_ID,
+                RoleReferenceType.ORGANIZATION
+            )
+        )
+            .thenReturn(Optional.ofNullable(role(API_PRIMARY_OWNER_ROLE_ID)));
+        when(
+            roleRepository.findByScopeAndNameAndReferenceIdAndReferenceType(
+                RoleScope.INTEGRATION,
+                SystemRole.PRIMARY_OWNER.name(),
+                ORG_ID,
+                RoleReferenceType.ORGANIZATION
+            )
+        )
+            .thenReturn(Optional.empty());
+
+        assertFalse(upgrader.upgrade());
```

Comment:
We can use assertj for all our asserts `assertThat(upgrader.upgrade()).isFalse)`